### ISSUE TITLE
Recently released Emacs 28.1 uses `C-x x` for buffer actions, thus conflicting with the prefix for perspective-el

### ISSUE THREAD

**Issue by paulophagula** [nex3/perspective-el#180]
Hi,

As the title says, recently released Emacs 28.1 uses `C-x x` for buffer actions, thus conflicting with the prefix for perspective-el.
The NEWS file says:

> A new keymap for buffer actions has been added.
The 'C-x x' keymap now holds keystrokes for various buffer-oriented
commands.  The new keystrokes are 'C-x x g' ('revert-buffer-quick'),
'C-x x r' ('rename-buffer'), 'C-x x u' ('rename-uniquely'), 'C-x x n'
('clone-buffer'), 'C-x x i' ('insert-buffer'), 'C-x x t'
('toggle-truncate-lines') and 'C-x x f' ('font-lock-update').

Should perpesctive-el switch to a new prefix? Is there a suggestion for another prefix we should use?

**Comment by gcv** [comment]
I have always considered `C-x x` an unfortunate default, and use `C-c C-p`. I don’t think it conflicts with anything popular..?

That said, it will be a breaking change in Perspective, something I try very hard to avoid. I suppose I can reduce its impact on existing users’ muscle memory by only changing the prefix for Emacs 28 and later.

**Comment by gcv** [comment]
@zenspider: Do you have an opinion on this? It's a helluva change to inflict on long-standing users of Perspective who never changed the prefix key. As far as I can tell, we can:

1. Ignore the new Emacs 28 bindings for `C-x x` and so shadow them.
2. Change the Perspective prefix, breaking users' muscle memory. Document in the changelog and hope that's enough. 
3. Option 2, but only for Emacs 28.

**Comment by zenspider** [comment]
I don't have an opinion... I immediately changed my prefix (^z ftw!) and suspect many people do.

Unfortunately we want to keep this around, otherwise we could lean into `declare` w/ `obsolete`. As such, I suspect option 3 is probably the most gentle, but option 2 could be fine as well and prepare people BEFORE they move over (so much stuff breaks on a big new release that it'd be nice not to be a part of that).

**Comment by gcv** [comment]
I just realized that `C-c C-p` conflicts with a bunch of stuff in Org, and also `markdown-mode`. I'm open to other suggestions.

**Comment by gcv** [comment]
What about `C-c M-p`?

**Comment by zenspider** [comment]
What about `nil`? Maybe don't set it and ask users to find something that works for them? I know I have mine set and thus won't even notice a version bump (nor will anyone else who already customized). Anyone who's using the default can have a message pop up or something.

**Comment by paulophagula** [comment]
Even though, it was I, who raised the issue, I'm keeping `C-x x` for perspective.el rather than the new buffer actions, as most of those buffer actions are in my particular view not so useful (except maybe for `rename-buffer`, but then again we have crux and so many hacks people made to rename a file along with its buffer). It is very seldom I'll want to perform any of the tasks listed in there.

**Comment by gcv** [comment]
@zenspider: Yes. Let's get out of the business of deciding on key prefixes for users. I looked at Projectile for inspiration, and it does not provide a default prefix.

@paulophagula: I agree with your reasoning about `C-x x`. It makes sense for users with muscle memory to keep it. Still, I think it's best for Perspective to just get out of the way and let users decide what to do.

I still don't want to kill it outright for everyone, so I kept `C-x x` or Emacs 27 and earlier, and blanked it for 28. There's a warning message for people who do not have the prefix set, and a way to disable the warning if someone wants the prefix off.

So overall, still a breaking change, but hopefully one easy to get past. I'd appreciate it if you two would give the change a quick look (see 7ec5a6ea0fc8ee8eade5731c3bfb20c52886d82d) and I'll merge it in a couple of days.
